sdk/apimanagement/Azure.ResourceManager.ApiManagement/src/autorest.md

```diff
@@ -321,6 +330,162 @@ directive:
     where: $.parameters
     transform: >
       $.OpenIdConnectIdParameter['x-ms-client-name'] = 'OpenId';
+      $.IfMatchOptionalParameter['x-ms-format'] = 'etag';
+      $.IfMatchRequiredParameter['x-ms-format'] = 'etag';
+  - from: apimgroups.json
+    where: $.paths
+    transform: >
+      $['/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/groups/{groupId}/users'].get.responses['200'].schema = {
+          "x-ms-client-name": "ApiManagementGroupUserListResult",
+          "type": "object",
+          "properties": {
+              "value": {
+                  "type": "array",
+                  "items": {
+                      "x-ms-client-name": "ApiManagementGroupUserData",
+                      "type": "object",
+                      "allOf": [
+                          {
+                              "$ref": "./definitions.json#/definitions/UserContract"
+                          }
+                      ]
```

Comment:
Why not just using a $ref here?

Reply:
The original swagger is just using a $ref here, this is going to cause the return type of this method to become the corresponding resource, but we found out that the returned id will not match the type of the resource, therefore I use an allOf to force a new model to be generated and thus bypass the act of wrapping the data model as a resource.